src/darwin/Framework/CHIP/MTRBaseSubscriptionCallback.h

```diff
@@ -93,6 +93,11 @@ class MTRBaseSubscriptionCallback : public chip::app::ClusterStateCache::Callbac
 
     chip::app::BufferedReadCallback & GetBufferedCallback() { return mBufferedReadAdapter; }
 
+    // Accessor for the cluster state cache, so that stale data can be cleared
+    // from it as needed.  The returned pointer must not be held long-term,
+    // because it can dangle if this callback object is destroyed.
+    chip::app::ClusterStateCache * GetClusterStateCache() { return mClusterStateCache.get(); }
```

Comment:
Providing access to the raw pointer seems to require more documentation about lifetime and provides a lot of usage. Any reason we did this instead of providing some `Clear` methods instead?

Reply:
I didn't feel it was worth the extra method definitions, and more importantly I could not figure out what to name the methods...  But I'm open to doing whatever here.

Reply:
Added Clear methods instead.  @andy31415 Please take a look when you get the chance?